functions/grafana/create_alert.py

```diff
@@ -0,0 +1,52 @@
+
+#
+# Copyright (c) 2024, RTE (http://www.rte-france.com)
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+
+import requests
+import constant
+from pathlib import Path
+import json
+
+#
+# @author Kevin Le Saulnier <kevin.lesaulnier at rte-france.com>
+#
+
+FOLDER_UID_PLACEHOLDER = "{{FOLDER_UID}}"
+DATASOURCE_UID_PLACEHOLDER = "{{DATASOURCE_UID}}"
+
+def create_alert_rule(alert_rule_json_path, parent_folder_uid, datasource_uid, rule_group_interval):
+    print("Importing alert-rule " + alert_rule_json_path)
+    file_content = Path(alert_rule_json_path).read_text()
+
+    file_content = file_content.replace(FOLDER_UID_PLACEHOLDER, parent_folder_uid)
+    file_content = file_content.replace(DATASOURCE_UID_PLACEHOLDER, datasource_uid)
+
+    file_content_json = json.loads(file_content)
+
+    try:
+        result = requests.post(constant.GRAFANA_ALERT_RULES, json = file_content_json, headers={'X-Disable-Provenance': 'true'})
```

Comment:
We should document maybe in the README? that we use X-Disable-Provenance': 'true' to make the alerts editable

Reply:
changed